### ISSUE TITLE
CMake files in (some?) libraries don't pick up include directories (e.g. servo2040)

### ISSUE THREAD

**Issue by ahnlak** [pimoroni/pimoroni-pico#826]
(from a question on Discord, but the deeper I dig the more confused I get!)

So, the Pico Boilerplate should allow us to pull in `pimoroni-pico` libraries simply by adding (for example):

```
include(libraries/servo2040/servo2040)
target_link_libraries(MyAwesomeApp servo2040)
```

BUT, if that library uses other libraries or drivers the provided `servo2040.cmake` does not expose the include directories of those sub-requirements.

While you can work around this by manually `include`ing all the drivers in your application's CMakeLists.txt that feels wrong (and involves a lot of digging around to find out what's required).

Some libraries (for example, `pico_explorer`) have suitable `include` directives to pick up the drivers that are required - my instinct is that this just got missed out in more recent libraries and, because the problem only really comes up when you're doing out-of-tree builds it probably doesn't upset the automated builds and until someone tries to use them in anger, we're none the wiser.

(I can't help feeling that there might be a more "CMake-y" way of doing it, but I haven't found it yet)

TL:DR - add `include` directories to library / driver cmake files to ensure they find required include files in out-of-tree builds.

I'm happy to go through and apply such a fix to places I can find, but as it involves touching a lot moving parts I thought I'd raise an issue first in case I've missed something obvious!

**Comment by Gadgetoid** [comment]
I distinctly remember previous a PR or Issue related to this somewhere, but I can't seem to dig it up.

At the time I think I hand-waived it away as needing some kind of "pimoroni_pico_init()" to set up all the paths so that everything would just work, but I think you make a good point- that some drivers set up include paths while others do not. Would make more sense - and be less complicated - to just fix this and make things consistent.
